src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventProvider.cs

```diff
@@ -883,25 +883,18 @@ private static unsafe object EncodeObject(ref object data, ref EventData* dataDe
             {
                 if (data is System.Enum)
                 {
-                    Type underlyingType = Enum.GetUnderlyingType(data.GetType());
```

Comment:
The fix in this file was prompted by testing where if you have enums that are not based on int or long, then it will serialize it as a string, but the manifest says it decodes like an enum.  This fix basically extends the types recognized to be all the integer types.  the 64 bit types are handled specially and all others are cast to int (32 bit case).